### ISSUE TITLE
[BUG] Torch dupe if there is no oxygen

### ISSUE THREAD

**Issue by AzazazChannel** [zmaster587/AdvancedRocketry#1751]
## Version of Advanced Rocketry
>AdvancedRocketry-1.12.2-1.6.0-190-universal

## Have you verified this is an issue in the latest unstable build
- [x] Y

## Version of LibVulpes
>LibVulpes-1.12.2-0.4.2-69-universal

## Version of Minecraft
>1.12.2 + forge 14.23.5.2838

## Does this occur without other mods installed
- [x] N

## Description of the problem
>When you set torch (Usual minecraft torch, yup) on space station **with oxygen**, all works fine (Torch is set and it dissapeared), but if you set it on station (maybe on planets too, but I hadn't tried it) **without oxygen**, it is being set, but torch doesn't disappear and you can set 2nd, 3rd etc,

**Comment by zmaster587** [comment]
It looks like what's happening is that the event the catches and replaces lit torches with unlit torches is called on the server before the call to place the block and decrement the stack count in the inventory is called.  So when the server goes to verify that the block can be placed, since the unlit torch is already there, the server thinks the block is occupied and doesn't decrement the count on the itemstack in the player's hand.  So when you go to pick up a torch again the server updates the inventory with the real count resulting in a dupe.

TL;DR, the server doesn't actually take the torch out of the inventory when placing it in the world and causes a client-server desync (even in single player).  Moving the stack in your inventory after placing the torch also forcibly syncs with the server.

**Comment by zmaster587** [comment]
Fixed for next release
